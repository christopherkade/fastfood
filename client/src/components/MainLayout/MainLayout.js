import React, { useState } from "react"
import styled from "styled-components"

import useWebsockets from "../../hooks/useWebsockets"
import { EmojiLayout } from "../EmojiLayout"
import { Link } from "../Link"
import { UserSelection } from "../UserSelection"
import { Button } from "../Button"
import { CompareModal } from "../CompareModal"
import { Notification } from "../Notification"

const Wrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
`

const TopWrapper = styled.div`
  padding: 5rem 10rem;

  @media (max-width: 1281px) {
    padding: 2rem 10rem;
  }

  @media (max-width: 1024px) {
    padding-bottom: 10rem;
  }

  @media (max-width: 769px) {
    padding: 2rem 1rem 10rem 1rem;
  }
`

const InfoWrapper = styled.div`
  display: flex;
  justify-content: space-between;

  @media (max-width: 1024px) {
    display: inline-block;
    text-align: center;
  }
`

const BottomWrapper = styled.div`
  width: 100%;
  bottom: 0;
  text-align: center;
  margin-top: 1rem;

  @media (max-width: 1024px) {
    position: fixed;
    bottom: 0.625rem;
  }
`

const Title = styled.h1`
  font-size: 1.5rem;
`

const MainLayout = () => {
  const [selections, setSelections] = useState(Array(4).fill({ name: "", emoji: "" }))
  const [compared, setCompared] = useState(false)
  const [{ link, sessionId, userId, foodCount, error }, { addFood }] = useWebsockets()

  /**
   * On food click, add an item to our selection array
   * If the array is full, reset it
   * @param {*} name
   * @param {*} emoji
   */
  const handleEmoijiClick = (name, emoji) => {
    setSelections((oldSelections) => {
      // Find the first empty item
      let index = oldSelections.findIndex((oldSelection) => oldSelection.name === "")

      if (index === -1) {
        oldSelections = Array(4).fill({ name: "", emoji: "" })
        index = 0;
      }
      oldSelections.splice(index, 1, { name, emoji })
      addFood([...oldSelections], sessionId, userId)
      return [...oldSelections]
    })
  }

  return (
    <Wrapper>
      <Notification error={error} />
      <TopWrapper>
        <InfoWrapper>
          <Title>Pick up to 4 types of food</Title>
          <Link link={link} />
        </InfoWrapper>
        <EmojiLayout onClick={handleEmoijiClick} />
      </TopWrapper>
      <BottomWrapper>
        <UserSelection selections={selections} />
        <Button onClick={() => setCompared(true)}>
          See results
        </Button>
      </BottomWrapper>
      {compared ? <CompareModal count={foodCount} onClick={() => setCompared(false)} /> : null}
    </Wrapper>
  )
}

export default MainLayout
