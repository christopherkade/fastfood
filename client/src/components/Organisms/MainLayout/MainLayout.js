import React, { useState } from "react"
import styled from "styled-components"

import useWebsockets from "hooks/useWebsockets"
import { EmojiLayout } from "components/Molecules/EmojiLayout"
import { Link } from "components/Atoms/Link"
import { Button } from "components/Atoms/Button"
import { CompareModal } from "components/Molecules/CompareModal"
import { Notification } from "components/Atoms/Notification"
import { Title } from "components/Atoms/Title"

const Wrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
`

const TopWrapper = styled.div`
  padding: 5rem 10rem;

  @media (max-width: 1281px) {
    padding: 2rem 10rem;
  }

  @media (max-width: 1024px) {
    padding-bottom: 10rem;
  }

  @media (max-width: 769px) {
    padding: 2rem 1rem 10rem 1rem;
  }
`

const InfoWrapper = styled.div`
  display: flex;
  justify-content: space-between;

  @media (max-width: 1024px) {
    display: inline-block;
    text-align: center;
  }
`

const BottomWrapper = styled.div`
  width: 100%;
  bottom: 0;
  text-align: center;
  position: fixed;
  bottom: 0.625rem;
`

const MainLayout = () => {
  const [selections, setSelections] = useState([])
  const [position, setPosition] = useState(0)
  const [compared, setCompared] = useState(false)
  const [{ link, sessionId, userId, foodCount, error }, { addFood }] = useWebsockets()

  /**
   * On food click, add an item to our selection array
   * If the array is full, reset it
   * @param {*} name
   * @param {*} emoji
   */
  const handleEmoijiClick = (name, emoji) => {
    setSelections((oldSelections) => {
      oldSelections.splice(position, 1, { name, emoji })
      addFood([...oldSelections], sessionId, userId)

      // When we have reached the maximum number of items, go back to 0, otherwise move to the next one
      position >= 3 ? setPosition(0) : setPosition(prevPosition => prevPosition + 1)

      return [...oldSelections]
    })
  }

  return (
    <Wrapper>
      <Notification error={error} />

      <TopWrapper>
        <InfoWrapper>
          <Title size="1.5">Pick up to 4 types of food</Title>
          <Link link={link} />
        </InfoWrapper>

        <EmojiLayout onClick={handleEmoijiClick} selections={selections} />
      </TopWrapper>

      <BottomWrapper>
        <Button onClick={() => setCompared(true)} isShaking={selections.length === 4}>
          See results
        </Button>
      </BottomWrapper>

      {compared ? <CompareModal count={foodCount} onClick={() => setCompared(false)} /> : null}
    </Wrapper>
  )
}

export default MainLayout
