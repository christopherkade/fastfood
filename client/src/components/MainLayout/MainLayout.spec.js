import React from 'react'
import { render, cleanup } from 'react-testing-library'

import MainLayout from "./MainLayout"

afterEach(cleanup);

describe("<MainLayout />", () => {
  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const { container } = render(<MainLayout />);
      expect(container).toMatchSnapshot();
    });
  })
})
