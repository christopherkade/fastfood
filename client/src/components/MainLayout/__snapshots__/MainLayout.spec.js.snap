// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainLayout /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-iwsKbI WwTTR"
  >
    <ul
      class="sc-bwzfXH dMyKrs"
    >
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Rice"
          role="img"
        >
          🍚
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Pizza"
          role="img"
        >
          🍕
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Burger"
          role="img"
        >
          🍔
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Sushi"
          role="img"
        >
          🍣
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Fries"
          role="img"
        >
          🍟
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Hotdog"
          role="img"
        >
          🌭
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Bacon"
          role="img"
        >
          🥓
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Eggs"
          role="img"
        >
          🥚
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Pancakes"
          role="img"
        >
          🥞
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Bread"
          role="img"
        >
          🍞
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Pretzel"
          role="img"
        >
          🥨
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Cheese"
          role="img"
        >
          🧀
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Salad"
          role="img"
        >
          🥗
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Burito"
          role="img"
        >
          🥙
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Egg Sandwich"
          role="img"
        >
          🥪
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Taco"
          role="img"
        >
          🌮
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Pork"
          role="img"
        >
          🍖
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Chicken"
          role="img"
        >
          🍗
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Steak"
          role="img"
        >
          🥩
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Shrimp"
          role="img"
        >
          🦐
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Fish"
          role="img"
        >
          🐟
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Crab"
          role="img"
        >
          🦀
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Octopus"
          role="img"
        >
          🐙
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Squid"
          role="img"
        >
          🦑
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Tomatoes"
          role="img"
        >
          🍅
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Mushrooms"
          role="img"
        >
          🍄
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Corn"
          role="img"
        >
          🌽
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Cucumber"
          role="img"
        >
          🥒
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Eggplant"
          role="img"
        >
          🍆
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Avocado"
          role="img"
        >
          🥑
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Potatoes"
          role="img"
        >
          🥔
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Broccoli"
          role="img"
        >
          🥦
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Carrots"
          role="img"
        >
          🥕
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Noodles"
          role="img"
        >
          🥡
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Bento Box"
          role="img"
        >
          🍱
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Riceball"
          role="img"
        >
          🍙
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Curry with Rice"
          role="img"
        >
          🍛
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Ramen"
          role="img"
        >
          🍜
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Dumpling"
          role="img"
        >
          🥟
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Mushroom Soup"
          role="img"
        >
          🍲
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Pasta"
          role="img"
        >
          🍝
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Paella"
          role="img"
        >
          🥘
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Cereal"
          role="img"
        >
          🥣
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Peanuts"
          role="img"
        >
          🥜
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Salted Popcorn"
          role="img"
        >
          🍿
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Pie"
          role="img"
        >
          🥧
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Ice-cream"
          role="img"
        >
          🍦
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Krispy Kreme"
          role="img"
        >
          🍩
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Chips Ahoy"
          role="img"
        >
          🍪
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Chocolate Cake"
          role="img"
        >
          🎂
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Strawberry Shortcake"
          role="img"
        >
          🍰
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Chocolate"
          role="img"
        >
          🍫
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Sweeties"
          role="img"
        >
          🍬
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Pudding"
          role="img"
        >
          🍮
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Kiwi"
          role="img"
        >
          🥝
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Grapes"
          role="img"
        >
          🍇
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Watermelon"
          role="img"
        >
          🍉
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Oranges"
          role="img"
        >
          🍊
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Bananas"
          role="img"
        >
          🍌
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Pineapples"
          role="img"
        >
          🍍
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Apples"
          role="img"
        >
          🍎
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Strawberries"
          role="img"
        >
          🍓
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Peaches"
          role="img"
        >
          🍑
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Pears"
          role="img"
        >
          🍐
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Cherries"
          role="img"
        >
          🍒
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Milk"
          role="img"
        >
          🥛
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Coffee"
          role="img"
        >
          ☕
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Tea"
          role="img"
        >
          🍵
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Coke"
          role="img"
        >
          🥤
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Beer"
          role="img"
        >
          🍺
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Wine"
          role="img"
        >
          🍷
        </span>
      </button>
      <button
        class="sc-bdVaJa iwkPNe"
      >
        <span
          aria-label="Apple Cider"
          role="img"
        >
          🍾
        </span>
      </button>
    </ul>
    <div
      class="sc-gZMcBi jwueps"
    >
      <div
        class="sc-EHOje YINzU"
      >
        <input
          class="sc-bZQynM ggbtGq"
          readonly=""
          value=""
        />
        <input
          class="sc-bZQynM ggbtGq"
          readonly=""
          value=""
        />
        <input
          class="sc-bZQynM ggbtGq"
          readonly=""
          value=""
        />
        <input
          class="sc-bZQynM ggbtGq"
          readonly=""
          value=""
        />
      </div>
      <button
        class="sc-gzVnrw cEPiAX"
      >
        <span
          aria-label=""
          role="img"
        >
          🍴
        </span>
         Compare 
        <span
          aria-label=""
          role="img"
        >
          🍴
        </span>
      </button>
      <div
        class="sc-htpNat dgczwS"
      >
        <p
          class="sc-bxivhb kXztPF"
        >
          Share this link with your friends!
        </p>
        <input
          class="sc-ifAKCX bfYiE"
          readonly=""
          value="Fetching link..."
        />
      </div>
    </div>
  </div>
</div>
`;
